lib/rspec/support/differ.rb

```diff
@@ -178,12 +181,7 @@ def object_to_string(object)
         object = @object_preparer.call(object)
         case object
         when Hash
-          object.keys.sort_by { |k| k.to_s }.map do |key|
-            pp_key   = PP.singleline_pp(key, "")
-            pp_value = PP.singleline_pp(object[key], "")
-
-            "#{pp_key} => #{pp_value},"
-          end.join("\n")
+          ObjectInspector.inspect(object)
```

Comment:
The are just called out with specific code for arrays, but I do believe arrays are rendered by the differ at times -- presumably by `pp`.  I'd have to experiment to see in which situations they get rendered and diffed but it definitely happens.